app/models/manageiq/providers/ibm_cloud/vpc/network_manager/cloud_subnet.rb

```diff
@@ -1,2 +1,43 @@
 class ManageIQ::Providers::IbmCloud::VPC::NetworkManager::CloudSubnet < ::CloudSubnet
+  include ProviderObjectMixin
+
+  supports :delete do
+    if ext_management_system.nil?
+      unsupported_reason_add(:delete, _("The subnet is not connected to an active %{table}") % {
+        :table => ui_lookup(:table => "ext_management_systems")
+      })
+    end
+    if number_of(:vms) > 0
+      unsupported_reason_add(:delete, _("The subnet has an active %{table}") % {
+        :table => ui_lookup(:table => "vm_cloud")
+      })
+    end
+  end
+
+  def delete_cloud_subnet_queue(userid)
+    task_opts = {
+      :action => "deleting Cloud Subnet for user #{userid}",
+      :userid => userid
+    }
+
+    queue_opts = {
+      :class_name  => self.class.name,
+      :method_name => 'raw_delete_cloud_subnet',
+      :instance_id => id,
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => 'ems_operations',
+      :zone        => ext_management_system.my_zone,
+      :args        => []
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
```

Comment:
This method should actually live in core since it isn't provider specific.  Can you add this to [`CloudSubnet`](https://github.com/ManageIQ/manageiq/blob/master/app/models/cloud_subnet.rb#L57) and remove the duplicates from
```
manageiq-providers-ibm_cloud/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/network_manager/cloud_subnet.rb:  def delete_cloud_subnet_queue(userid)
manageiq-providers-nuage/app/models/manageiq/providers/nuage/network_manager/cloud_subnet.rb:  def delete_cloud_subnet_queue(userid)
manageiq-providers-openstack/app/models/manageiq/providers/openstack/network_manager/cloud_subnet.rb:  def delete_cloud_subnet_queue(userid)
```